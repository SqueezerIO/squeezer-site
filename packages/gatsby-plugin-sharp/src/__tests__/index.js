const path = require(`path`)

const { base64, responsiveSizes } = require(`../`)

describe(`gatsby-plugin-sharp`, () => {
  const args = {
    duotone: false,
    grayscale: false,
    rotate: false,
  }
  const absolutePath = path.join(__dirname, `images/test.png`)
  const file = getFileObject(absolutePath)

  describe(`responsiveSizes`, () => {
    it(`includes responsive image properties, e.g. sizes, srcset, etc.`, async () => {
      const result = await responsiveSizes({ file })

      expect(result).toMatchSnapshot()
    })

    it(`adds pathPrefix if defined`, async () => {
      const pathPrefix = `/blog/posts`
      const result = await responsiveSizes({
        file,
        args: {
          pathPrefix,
        },
      })

      expect(result.src.indexOf(pathPrefix)).toBe(0)
      expect(result.srcSet.indexOf(pathPrefix)).toBe(0)
    })

    it(`accounts for pixel density`, async () => {
      const result = await responsiveSizes({
        file: getFileObject(path.join(__dirname, `images/144-density.png`)),
      })

      expect(result).toMatchSnapshot()
    })

    it(`does not change the arguments object it is given`, async () => {
      const args = { maxWidth: 400 }
      await responsiveSizes({
        file,
        args,
      })

      expect(args).toEqual({ maxWidth: 400 })
    })
  })

  describe(`base64`, () => {
    it(`converts image to base64`, async () => {
      const result = await base64({
        file,
        args,
      })

      expect(result).toMatchSnapshot()
    })
  })
})

function getFileObject(absolutePath) {
  return {
    id: `${absolutePath} absPath of file`,
    absolutePath,
    extension: `png`,
    internal: {
      contentDigest: `1234`,
    },
  }
}
