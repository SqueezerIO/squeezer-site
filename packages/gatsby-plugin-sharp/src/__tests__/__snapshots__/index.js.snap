// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-sharp base64 converts image to base64 1`] = `
Object {
  "aspectRatio": 1,
  "height": 20,
  "originalName": undefined,
  "src": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAGElEQVQ4y2P4TwFgGNU8qnlU86jmgdUMAOBEq42KgAyMAAAAAElFTkSuQmCC",
  "width": 20,
}
`;

exports[`gatsby-plugin-sharp responsiveSizes accounts for pixel density 1`] = `
Object {
  "aspectRatio": 2.0661764705882355,
  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAKCAYAAAC0VX7mAAAACXBIWXMAABYlAAAWJQFJUiTwAAAA9klEQVQoz7WRQUvDQBCF+/9v3sSzl+K9oqStVqmoN2mpJLZJYyweKpiQbLLZJM83JQ1VCrnEheF9zA5vZ2Z76Pj0/s3QVNVOS6oqyx1r8j5vDriqdV9/1PBRKTha4y5JcBMnsMn3icKI/EaeMC+8JMu9FccNz7MMRW3cGAZ5jm1RwKNKbE0Bn+rqHF/Mr8mrmkXlEWG5/zSmmaqzHeq/HXb6KbLknCNl3IVoazccN2VtGEX4DiOEjOqwQ5Wm8Px3vDoOlq7bahhsNqzzcGkNMX16hjW+/W3oBx+YTB9wNRxjcG21Gq68NRa2jZPTM7zM5jjvXzSGP/5nCvc9Ga2VAAAAAElFTkSuQmCC",
  "density": 144,
  "originalImg": "/static/1234-f54e4.png",
  "originalName": undefined,
  "presentationHeight": 68,
  "presentationWidth": 141,
  "sizes": "(max-width: 141px) 100vw, 141px",
  "src": "/static/1234-f54e4.png",
  "srcSet": "/static/1234-a0d9a.png 200w,
/static/1234-f54e4.png 281w",
}
`;

exports[`gatsby-plugin-sharp responsiveSizes includes responsive image properties, e.g. sizes, srcset, etc. 1`] = `
Object {
  "aspectRatio": 1,
  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAGElEQVQ4y2P4TwFgGNU8qnlU86jmgdUMAOBEq42KgAyMAAAAAElFTkSuQmCC",
  "density": 72,
  "originalImg": "/static/1234-18c2a.png",
  "originalName": undefined,
  "presentationHeight": 1,
  "presentationWidth": 1,
  "sizes": "(max-width: 1px) 100vw, 1px",
  "src": "/static/1234-18c2a.png",
  "srcSet": "/static/1234-18c2a.png 1w",
}
`;
